### PR TITLE
EE-235 Clear all notifications swagger

### PR DESCRIPTION
**Pull request by Maxiweb**
https://io1dev.atlassian.net/browse/EE-235

### CODE REVIEW

File: api.swagger.yaml [hunk 1]
```diff
@@ -5471,6 +5471,16 @@ paths:
...
             The notification does not exist.
       description: Mark a notification as read.
+  '/notifications/clear_all':
+    delete:
+      tags:
+        - notifications
+      responses:
+        '200':
+          description: The user marked the notification as read.
+        '400':
```
Comment (ParryLuke) [inline api.swagger.yaml]: Do we know of any error responses that we can expect to be more specific?

File: api.swagger.yaml [hunk 2]
```diff
@@ -5471,6 +5471,16 @@ paths:
           description: |
             The notification does not exist.
       description: Mark a notification as read.
+  '/notifications/clear_all':
+    delete:
+      tags:
+        - notifications
+      responses:
+        '200':
+          description: The user marked the notification as read.
```
Comment (ParryLuke) [inline api.swagger.yaml]: notifications*

File: api.swagger.yaml [hunk 3]
```diff
@@ -5471,6 +5471,16 @@ paths:
           description: |
             The notification does not exist.
       description: Mark a notification as read.
+  '/notifications/clear_all':
+    delete:
```
Comment (ParryLuke) [inline api.swagger.yaml]: Should be a PUT request. The notifications don't actually get deleted, just marked as read

File: api.swagger.yaml (hunk 1 above)
Comment (Maxiweb) [reply api.swagger.yaml]: Not as of yet. I will probably update onces i have finished all tasks in the epic.

File: api.swagger.yaml (hunk 2 above)
Comment (ParryLuke) [reply api.swagger.yaml]: This still needs to be updated

File: api.swagger.yaml (hunk 1 above)
Comment (ParryLuke) [reply api.swagger.yaml]: Can any of these be added?

File: api.swagger.yaml [hunk 4]
```diff
@@ -5471,6 +5471,23 @@ paths:
...
+        in: query
+        name: app_notifications_only
+        schema:
+          type: boolean
+          default: false
+    put:
+      tags:
+        - notifications
+      responses:
+        '200':
```
Comment (jack-durrant) [inline api.swagger.yaml]: Please document the expected response body.

File: api.swagger.yaml [hunk 5]
```diff
@@ -5471,6 +5471,29 @@ paths:
...
       description: Mark a notification as read.
+  '/notifications/clear_all':
+    parameters:
+      - description: Marks all notifications for the mobile app as read. If true, this will mark as read notifications with entity type user, and entity type node and bundle type post only.
+        in: query
+        name: app_notifications_only
+        schema:
+          type: boolean
+          default: false
+    put:
```
Comment (ParryLuke) [inline api.swagger.yaml]: Can this be a POST request? sorry

File: api.swagger.yaml [hunk 6]
```diff
@@ -5471,6 +5471,29 @@ paths:
...
+      tags:
+        - notifications
+      responses:
+        '200':
```
Comment (ParryLuke) [inline api.swagger.yaml]: And can this be a 204 response? We can remove the response body and related schema

File: api.swagger.yaml [hunk 7]
```diff
@@ -5490,6 +5490,8 @@ paths:
               schema:
                 $ref: '#/components/schemas/notifications-clear-all-schema'
         '400':
+          description: Could not find user to corresponding uuid.
```
[the remark this answers is not in the document]
Comment (Maxiweb) [reply api.swagger.yaml]: Updated to 404  
We get uuid from userData in node which then gets passed to drupal services which checks it matches logged in user etc.
const { userId } = req.userData;

File: api.swagger.yaml [hunk 8]
```diff
@@ -5471,6 +5471,25 @@ paths:
...
+      tags:
+        - notifications
+      responses:
+        '204':
+          description: The user marked the notifications as read.
+        '403':
+          description: User is not authorized to clear all notifications.
+        '404':
+          description: Could not find user to corresponding uuid.
+      description: Clear all notifications for a particular user.
```
Comment (jmorahan) [inline api.swagger.yaml]:
```suggestion
      description: Clear all notifications for the current user.
```

File: api.swagger.yaml [hunk 9]
```diff
@@ -5471,6 +5471,25 @@ paths:
...
+    post:
+      tags:
+        - notifications
+      responses:
+        '204':
+          description: The user marked the notifications as read.
+        '403':
+          description: User is not authorized to clear all notifications.
+        '404':
+          description: Could not find user to corresponding uuid.
```
Comment (jmorahan) [inline api.swagger.yaml]: This doesn't take a UUID parameter so does it really make sense to return a 404 here? If somehow the _current_ user doesn't exist, wouldn't that just be a 403?

**Pull request merged**